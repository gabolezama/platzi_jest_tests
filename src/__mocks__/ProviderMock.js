import React from "react";
import { createStore } from "redux";
import { Router } from "react-router-dom";
import { Provider } from "react-redux";
import { createBrowserHistory } from "history";
import reducer from "../reducers";
import initialState from "../initialState";

const store = createStore( reducer, initialState);
const history = createBrowserHistory();

export const ProviderMock = props =>(
    <Provider store={store}>
        <Router history={history}>
            {props.children}
        </Router>
    </Provider>
)