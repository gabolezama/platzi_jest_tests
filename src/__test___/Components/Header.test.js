import React from "react";
import { mount, render, shallow } from "enzyme";
import { ProviderMock } from "../../__mocks__/ProviderMock";
import Header from "../../components/Header";

describe("<Header/>", function(){
    test('Render del Header', () => {
        const header = shallow(
            <ProviderMock>
                <Header/>
            </ProviderMock>
        );
        expect(header.length).toEqual(1);
    })

    test('Render del titulo', () => {
        const header = mount(
            <ProviderMock>
                <Header/>
            </ProviderMock>
        );
        expect(header.find(".Header-title").text()).toEqual("Platzi Store");
    })
})

describe("<Header/> Snapshot", function(){
    test('Probar el snapshot de Header', () => {
        const header = render(
            <ProviderMock>
                <Header/>
            </ProviderMock>
        );
        expect(header).toMatchSnapshot();
    })
})