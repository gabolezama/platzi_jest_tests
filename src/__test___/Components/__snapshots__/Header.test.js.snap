// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> Snapshot Probar el snapshot de Header 1`] = `
LoadedCheerio {
  "0": <div
    class="Header"
  >
    <h1
      class="Header-title"
    >
      <a
        href="/"
      >
        Platzi Store
      </a>
    </h1>
    <div
      class="Header-checkout"
    >
      <a
        href="/checkout"
      >
        <i
          class="fas fa-shopping-basket"
        />
      </a>
    </div>
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
